modules/transaction_payment/src/lib.rs

```diff
@@ -694,4 +698,65 @@ pub mod module {
 			Ok(())
 		}
 	}
+
+	impl<T: Config + Send + Sync> EnsureCanChargeFee<T::AccountId, PalletBalanceOf<T>, NegativeImbalanceOf<T>>
+		for ChargeTransactionPayment<T>
+	where
+		PalletBalanceOf<T>: Send + Sync + FixedPointOperand,
+	{
+		fn reserve_fee(who: &T::AccountId, weight: Weight) -> DispatchResult {
+			let fee = Module::<T>::weight_to_fee(weight);
+			Module::<T>::ensure_can_charge_fee(who, fee, WithdrawReasons::TRANSACTION_PAYMENT);
+			<T as Config>::Currency::reserve(&who, fee)
+		}
+
+		fn unreserve_and_charge_fee(
+			who: &T::AccountId,
+			weight: Weight,
+		) -> Result<(PalletBalanceOf<T>, Option<NegativeImbalanceOf<T>>), TransactionValidityError> {
+			let fee = Module::<T>::weight_to_fee(weight);
+			<T as Config>::Currency::unreserve(&who, fee);
+			Module::<T>::ensure_can_charge_fee(who, fee, WithdrawReasons::TRANSACTION_PAYMENT);
```

Comment:
I don't think we want this again. The unreserve should make sure there is enough ACA and ensure_can_charge_fee could buy more ACA if the preferred token is not ACA